### PR TITLE
Add unittests for RemoteClusterStateAttributesManager

### PR DESCRIPTION
**Pull request by shiv0408**
### Description
Add unit test for RemoteClusterStateAttributesManager for write flow and failure case

### Check List
- [x] Functionality includes testing.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by shiv0408** [comment]
> ❌ Gradle check result for https://github.com/opensearch-project/OpenSearch/commit/3d3f2c52f52829007275ce52e9e00fa0f96c6248: [FAILURE](https://build.ci.opensearch.org/job/gradle-check/41310/)

`org.opensearch.indices.IndicesRequestCacheIT.testStaleKeysRemovalWithoutExplicitThreshold {p0={"search.concurrent_segment_search.enabled":"false"}}` - #14288

**Comment by dblock** [comment]
Thanks! Maybe @shwetathareja can CR?

File: test/framework/src/main/java/org/opensearch/common/util/TestCapturingListener.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+package org.opensearch.common.util;
+
+import org.opensearch.core.action.ActionListener;
+
+/**
+ * A simple implementation of {@link ActionListener} that captures the response and failures used for testing purposes.
+ *
+ * @param <T> the result type
+ */
+public class TestCapturingListener<T> implements ActionListener<T> {
```
Comment (shwetathareja) [inline test/framework/src/main/java/org/opensearch/common/util/TestCapturingListener.java:18]: it can be local to the your test class

Comment (shiv0408) [reply test/framework/src/main/java/org/opensearch/common/util/TestCapturingListener.java:18]: This was required for other test files as well, I added this in another PR https://github.com/opensearch-project/OpenSearch/pull/14394 as well, which is merged. I haven't rebased this that's why this is coming up in the diff here.

Comment (shwetathareja) [reply test/framework/src/main/java/org/opensearch/common/util/TestCapturingListener.java:18]: ok please rebase

Comment (shiv0408) [reply test/framework/src/main/java/org/opensearch/common/util/TestCapturingListener.java]: done

**Review by shwetathareja — APPROVED** [review]

**Pull request merged**